source/server/config/network/client_ssl_auth.cc

```diff
@@ -1,34 +1,50 @@
+#include "client_ssl_auth.h"
+
 #include "envoy/network/connection.h"
 #include "envoy/server/instance.h"
 
 #include "common/filter/auth/client_ssl.h"
-#include "server/configuration_impl.h"
 
 namespace Server {
 namespace Configuration {
 
-/**
- * Config registration for the client SSL auth filter. @see NetworkFilterConfigFactory.
- */
-class ClientSslAuthConfigFactory : public NetworkFilterConfigFactory {
-public:
-  // NetworkFilterConfigFactory
-  NetworkFilterFactoryCb tryCreateFilterFactory(NetworkFilterType type, const std::string& name,
-                                                const Json::Object& json_config,
-                                                Server::Instance& server) {
-    if (type != NetworkFilterType::Read || name != "client_ssl_auth") {
-      return nullptr;
-    }
-
-    Filter::Auth::ClientSsl::ConfigPtr config(Filter::Auth::ClientSsl::Config::create(
-        json_config, server.threadLocal(), server.clusterManager(), server.dispatcher(),
-        server.stats(), server.random()));
-    return [config](Network::FilterManager& filter_manager) -> void {
-      filter_manager.addReadFilter(
-          Network::ReadFilterPtr{new Filter::Auth::ClientSsl::Instance(config)});
-    };
+const std::string ClientSslAuthConfigFactory::CLIENT_SSL_SCHEMA(R"EOF(
+  {
+    "$schema": "http://json-schema.org/schema#",
+    "properties": {
+      "auth_api_cluster" : {"type" : "string"},
+      "stat_prefix" : {"type" : "string"},
+      "ip_white_list" : {
+        "type" : "array",
+        "items" : {
+          "type": "string",
+          "format" : "ipv4"
+        }
+      }
+    },
+    "required": ["auth_api_cluster", "stat_prefix"],
+    "additionalProperties": false
```

Comment:
Unfortunately, there is no way to set it globally for schemas in rapidjson or in JSON schema.